src/util.ts

```diff
@@ -30,3 +30,34 @@ export async function exhaust(iterable: AsyncIterable<unknown>) {
         }
     }
 }
+
+/**
+ * Creates a `next` function and associated promise to promise-ify a node style callback. The
+ * `next` function must be passed as the callback to a function, and the resulting error or value
+ * will be emitted from the promise. The promise will always resolve.
+ *
+ * The error value of the callback (first parameter) will be emitted as an `Error` atom from the
+ * promise, whilst the value of the callback (second parameter) will be emitted as an `Ok` atom on
+ * the promise.
```

Comment:
Thoughts on this wording?


Creates a promise and `next` callback pair. When `next` is called as a node-style callback with an error or value, the promise will be resolved with either an Okay atom or an Error atom. This can be used to promise-ify a node style callback. When next is called, the promise is guaranteed to be resolved.

```ts
@example
const [promise, next] = createNodeCallback();
promise.then(console.log)
next(new Error("Oh no!"), null)
// console.logs an Error Atom with value=Error("Oh no")
```

```ts
@example
const [promise, next] = createNodeCallback();
promise.then(console.log)
next(null, true)
// console.logs an Okay Atom w/ value=true
```